daffodil-test/src/test/resources/org/apache/daffodil/section07/discriminators/multipleDiscriminators.tdml

```diff
@@ -0,0 +1,171 @@
+  </tdml:parserTestCase>
+  
+  <!--
+    The first discriminator evaluates to false, causing immediate backtracking.
+    The second discriminator is never evaluated.
+    We speculatively parse innerBranch2, and it is successful.
+    Exactly he same behavior as multipleDiscriminators1.
+    -->
+  <tdml:parserTestCase name="multipleDiscriminators2" model="multipleDiscriminators" root="root">
+    <tdml:document>fta</tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <root>
+          <discrim1>f</discrim1>
+          <discrim2>t</discrim2>
+          <outerBranch1>
+            <innerBranch2>a</innerBranch2>
+          </outerBranch1>
+        </root>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <!--
+    The first discriminator evaluates to true, resolving the inner choice PoU.
+    The second discriminator evaluates to false, causing immediate backtracking.
+    Because the inner choice PoU is resolved, we backtrack to the outer choice.
+    We speculatively parse outerBranch2 and it is successful.
+  -->
+  <tdml:parserTestCase name="multipleDiscriminators3" model="multipleDiscriminators" root="root">
+    <tdml:document>tfa</tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <root>
+          <discrim1>t</discrim1>
+          <discrim2>f</discrim2>
+          <outerBranch2>a</outerBranch2>
+        </root>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <!--
+    The first discriminator evaluates to true, resolving the inner choice PoU.
+    The second discriminator evaluates to true, resolving the outer choice PoU.
+    We successfully parse the integer element. No backtracking is neeeded.
+  -->
+  <tdml:parserTestCase name="multipleDiscriminators4" model="multipleDiscriminators" root="root">
+    <tdml:document>tt1</tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <root>
+          <discrim1>t</discrim1>
+          <discrim2>t</discrim2>
+          <outerBranch1>
+            <innerBranch1>
+              <integer>1</integer>
+            </innerBranch1>
+          </outerBranch1>
+        </root>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <!--
+    The first discriminator evaluates to true, resolving the inner choice PoU.
+    The second discriminator evaluates to true, resolving the outer choice PoU.
+    We attempt to parse the interger element, which fails, creating a Parse Error and immediately backtrack.
+    Because the inner choice PoU is resolved, we backtrack to the outer choice.
+    Because the outer choice PoU is resolved, we backtrack again, but there are
+    no more PoU's. It is a Parse Error related to the failure to parse the
+    interger element.
+  -->
+  <tdml:parserTestCase name="multipleDiscriminators5" model="multipleDiscriminators" root="root">
+    <tdml:document>tta</tdml:document>
+    <tdml:errors>
+      <tdml:error>interger</tdml:error>
```

Comment:
Is "interger" spelled correctly?  A grep shows me no occurrences of "interger" in my incubator-daffodil checkout and a find in page tells me "interger" occurs only 3 times in this page (two times in the test case comment, one time inside the test case).  Since I see an "interger" element defined in the test schema, I assume you meant "integer" instead of "interger".

Reply:
Yep, that should be "integer" to match the element name related to the parser error. Good catch. Will fix.